llm4pddl/envs/assets/pddl/dressed/generate_tasks.py

```diff
@@ -0,0 +1,199 @@
+                          num_formal_in_suit)
+    # deciding extra clothing:
+    extras = ['ex_p', 'ex_d', 'ex_sp', 'ex_ss', 'ex_np', 'ex_cs', 'ex_sj']
+    exs = {ex: 0 for ex in extras}
+    for _ in range(num_extra_objects):
+        exs[rng.choice(extras)] += 1
+    # making objects:
+    persons = [f'person{i}' for i in range(1, num_people + 1 + exs['ex_p'])]
+    dresses = [
+        f'dress{i}' for i in range(1, num_formal_in_dress + 1 + exs['ex_d'])
+    ]
+    sweatpants = [
+        f'sweatpants{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_sp'])
+    ]
+    sweatshirts = [
+        f'sweatshirt{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_ss'])
+    ]
+    nicepants = [
+        f'nice-pants{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_np'])
+    ]
+    collaredshirt = [
+        f'collared-shirt{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_cs'])
+    ]
+    suitjacket = [
+        f'suit-jacket{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_d'])
+    ]
+    objects = [{
+        'type': 'person',
+        'objs': persons
+    }, {
+        'type': 'dress',
+        'objs': dresses
+    }, {
+        'type': 'sweatpants',
+        'objs': sweatpants
+    }, {
+        'type': 'sweatshirt',
+        'objs': sweatshirts
+    }, {
+        'type': 'nice-pants',
+        'objs': nicepants
+    }, {
+        'type': 'collared-shirt',
+        'objs': collaredshirt
+    }, {
+        'type': 'suit-jacket',
+        'objs': suitjacket
+    }]
+    object_str = ''
+    for obj in objects:
+        if len(obj['objs']) > 0:
+            object_str += ' '.join(
+                obj['objs']) + f' - {obj["type"]}\n            '
+    # making initial predicates:
+    init_preds = ''
+    for person in persons:
+        init_preds += (f'(wearing-nothing-formal {person})\n         ' +
+                       f'(wearing-nothing-casual {person})\n         ')
+    for clothing in (dresses + sweatpants + sweatshirts + nicepants +
+                     collaredshirt + suitjacket):
+        init_preds += f'(in-closet {clothing})' + '\n         '
+    # making goal predicates:
+    goals = ''
+    goal_predicates = []
+    rng.shuffle(persons)
+    for i in range(num_casual_events):
+        goal_predicates.append(
+            f'(attending-casual-event {persons[i]})\n              ')
+    for i in range(
+            num_casual_events,
+        (num_casual_events + num_formal_in_dress + num_formal_in_suit)):
+        goal_predicates.append(
+            f'(attending-formal-event {persons[i]})\n              ')
+    rng.shuffle(goal_predicates)
+    for goal in goal_predicates:
+        goals += goal
+    #making problem:
+    problem_string = f"""(define (problem dressed)
+  (:domain dressed)
+  (:objects {object_str})
+  (:init {init_preds})
+  (:goal (and {goals}))
+  )"""
+    return problem_string
+
+
+def test_generate_dressing_problems() -> None:
```

Comment:
love that you added tests!! let's move these into the top level `tests/` directory so that they get run automatically by the github checks. so, make some new directories and files: `llm4pddl/tests/envs/assets/pddl/dressed/test_generate_tasks.py` and move these tests in there. to make sure that everything works, you can then do `pytest -s llm4pddl/tests/envs/assets/pddl/dressed/test_generate_tasks.py`.

Reply:
I just did this, and now it says the unittests dont pass for the new commit even though the command you provided does work. Did I do something wrong in the new version I just pushed?

Reply:
I'm looking at the unit test check on the last commit (clicking "Details" next to llm4pddl / unit-tests (3.8) ... below). Is that what you're referring to? It looks like the tests pass, but it's saying that not all lines are covered, that is, not all of the code that you wrote in the main file gets executed by your tests. Take a look at the lines that it's indicating are uncovered, and see if you can easily update your test to cover those lines. (If not, I can help!)